cubequery/api_server.py

```diff
@@ -107,9 +111,28 @@ def all_tasks():
     return jsonify(normalise_task_info(result))
 
 
+@app.route('/token', methods=['POST'])
+def get_token():
+    payload = request.get_json()
+    if not payload['name']:
+        abort(403, "name required")
+
+    if not payload['pass']:
+        abort(403, "pass required")
+
+    logging.info(f"log in request for {payload['name']} from {request.remote_addr}")
+
+    if not users.check_user(payload['name'], payload['pass'], request.remote_addr):
+        abort(403, "bad creds")
+
+    s = Serializer(get_config("App", "secret_key"), expires_in=600)
+    return jsonify({'token': s.dumps({'id': payload['name']})})
+
+
 @app.route('/task', methods=['POST'])
 def create_task():
-    if not validate_app_key(request):
+    valid, user_id = validate_app_key(request)

```

Comment:
you have called this same function 4 times it would be better to add this as a separate function and throw the abort you could also then include the reason for this.. (not valid/ expired)